util/io: fix resize of mmap-ed files

ret2libc opened the pull request:
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

When resizing files through `r` commands, although the buffers seemed to be resized while the file was opened, the files were not really truncated. Thus when re-opening the written file, no actual change was made.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

See test included in PR.

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

ret2libc committed:
util: fix resize of mmap-ed files

When resizing files through `r` commands, although the buffers seemed to
be resized while the file was opened, the files were not really
truncated. Thus when re-opening the written file, no actual change was
made.

wargio commented:
```
test_rz_buf_mmap ERR
[XX] Fail at line 26: New content should be written

ERR
[XX] Fail at line 169: test failed

test_rz_buf_with_buf OK
test_rz_buf_slice OK
test_rz_buf_io_fd ERR
[XX] Fail at line 26: New content should be written

ERR
[XX] Fail at line 223: test failed
```

XVilka commented:
```

[XX] db/cmd/cmd_r extend file
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc 'o+ .tmp_r_file
r
w ABC
r+ 3
w ABC
ps 3
r 0x40
r
wb 4142434445
r- 4
ps 5
o- 3
o .tmp_r_file
r
o- 3
rm .tmp_r_file
' =
-- stderr
--- expected
+++ actual
@@ -1,1 +1,2 @@
+../librz/util/buf_bytes.c:79:2: runtime error: null pointer passed as argument 2, which is declared to never be null
 ERROR: Could not write 'ABC' at 0
```

ret2libc committed:
io: fix iowhence2buf function

The function should convert the IO-specific whence enum to the
RzBuffer-specific whence enum, however it was doing the opposite.
Fortunately, it doesn't really affect anything because those two enums
have the same values.

ret2libc committed:
db/cmd/cmd_r: do not hardcode the fd num

ret2libc committed:
util/buf: avoid undefined behaviour when dealing with empty files
